### PR TITLE
Added cython3 support and support for staticmethod decorator

### PR DESCRIPTION
**Pull request by jpfeuffer**
As you can see in the added tests, Cython3 does not add a member function anymore to attached free functions
and therefore cannot be called with objects anymore.
This might break tests in OpenMS but we should switch to @staticmethod declarators anyway.

### CODE REVIEW

File: tests/test_code_generator.py [hunk 1]
```diff
@@ -189,9 +189,18 @@ def test_templated():
...
+    if int(str(cython_version).split(".")[0]) < 3:
+        assert templated.computeEight() == 8
+        assert templated_o.computeEight() == 8
+    assert twrapped.Templated.computeEight() == 8
+    assert twrapped.Templated_other.computeEight() == 8
+
+    # Test static functions (can be called with or without object)
     assert templated.computeSeven() == 7
     assert templated_o.computeSeven() == 7
+    assert twrapped.Templated.computeSeven() == 7
```
Comment (timosachsenberg) [inline tests/test_code_generator.py:202]: ok this is really cool but do we need a check for the cython version here as well?.

File: autowrap/DeclResolver.py [hunk 2]
```diff
@@ -596,7 +597,8 @@ def _resolve_method_or_function(method_decl, instance_mapping, local_type_map,
         arg_type = _resolve_alias(arg_type, instance_mapping, local_type_map)
         args.append((arg_name, arg_type))
     name = method_decl.annotations.get("wrap-as", method_decl.name)
-    return clz(name, result_type, args, method_decl, instance_mapping, local_type_map)
+
+    return clz(name, method_decl.is_static, result_type, args, method_decl, instance_mapping, local_type_map)
```
Comment (timosachsenberg) [inline autowrap/DeclResolver.py:601]: add comment what clz means?

**Review by timosachsenberg — APPROVED** [review]

**Comment by timosachsenberg** [comment]
if we move to cython 3 it would also be good to add: https://cython.readthedocs.io/en/latest/src/userguide/wrapping_CPlusPlus.html#scoped-enumerations

**Comment by timosachsenberg** [comment]
do you know if this affects https://github.com/OpenMS/autowrap/issues/105 ?

**Review by axelwalter — APPROVED** [review]

File: tests/test_code_generator.py (hunk 1 above)
Comment (jpfeuffer) [reply tests/test_code_generator.py:202]: No, apparently this would have worked for a long time already. Also with Cython 0. See the test matrix.

**Comment by jpfeuffer** [comment]
Hmm so I checked this yesterday and spent way too much time on trying to figure it out.
Yes, you can declare `cpdef enum class` now, but I did not find a way to use correctly with autowrap yet.
See https://groups.google.com/g/cython-users/c/PpwhyIzqGyA
It definitely would need changes in autowrap since we currently treat it as a normal enum and normal enums can be implicitly converted to ints (which we rely on).

**Comment by jpfeuffer** [comment]
Ok I think I found a way. I guess this is still a bit experimental.

File: autowrap/DeclResolver.py (hunk 2 above)
Comment (jpfeuffer) [reply autowrap/DeclResolver.py:602]: clz is a param of the function. There are too many instances of this kind of usage.
There should be a big PR with typing support.

**Pull request merged**